frontend/packages/dev-console/src/components/catalog/providers/useDevfileSamples.tsx

```diff
@@ -0,0 +1,57 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { APIError } from '@console/shared';
+import { ExtensionHook, CatalogItem } from '@console/dynamic-plugin-sdk';
+import { coFetchJSON } from '@console/internal/co-fetch';
+import { DevfileSample } from '../../import/devfile/devfile-types';
+
+const normalizeDevfileSamples = (devfileSamples: DevfileSample[]): CatalogItem[] => {
+  const normalizedDevfileSamples = _.map(devfileSamples, (sample) => {
+    const { name: uid, displayName, description, tags, git, icon } = sample;
+    const gitRepoUrl = Object.values(git.remotes)[0];
+    const iconUrl = icon ? `data:image/png;base64,${icon}` : '';
+    const href = `/import?importType=devfile&formType=sample&devfileName=${uid}&gitRepo=${gitRepoUrl}`;
+
+    const item: CatalogItem = {
+      uid,
+      type: 'Sample',
+      name: displayName,
+      description,
+      tags,
+      cta: {
+        label: '',
```

Comment:
This will require a label to work with @invincibleJai 's change to include samples in quick search.

Reply:
Added label.